components/ws-daemon/pkg/daemon/daemon.go

```diff
@@ -77,14 +77,24 @@ func NewDaemon(config Config, reg prometheus.Registerer) (*Daemon, error) {
 		cgroupV2IOLimiter,
 		&cgroup.ProcessPriorityV2{
 			ProcessPriorities: map[cgroup.ProcessType]int{
-				cgroup.ProcessSupervisor: -10,
+				cgroup.ProcessWorkspaceKit: -10,
+				cgroup.ProcessSupervisor:   -10,
 
 				cgroup.ProcessIDE:          -10,
 				cgroup.ProcessWebIDEHelper: -5,
 
 				cgroup.ProcessCodeServer:       -10,
 				cgroup.ProcessCodeServerHelper: -5,
 			},
+			EnableOOMScoreAdj: config.OOMScores.Enabled,
+			OOMScoreAdj: map[cgroup.ProcessType]int{
+				cgroup.ProcessWorkspaceKit:     config.OOMScores.Tier1,
+				cgroup.ProcessSupervisor:       config.OOMScores.Tier1,
+				cgroup.ProcessCodeServer:       config.OOMScores.Tier1,
+				cgroup.ProcessIDE:              config.OOMScores.Tier1,
+				cgroup.ProcessCodeServerHelper: config.OOMScores.Tier2,
+				cgroup.ProcessWebIDEHelper:     config.OOMScores.Tier2,
```

Comment:
@Furisto may I ask, did you socialize this improvement with the IDE team? I ask because it would make sense to get their feedback on this grouping and tier assignment.

Reply:
@gitpod-io/engineering-ide What do you think about this assignment? Tier 1 processes would not be oom killed with this change and for Tier 2 processes it is less likely. You can see how we identity the processes [here.](https://github.com/gitpod-io/gitpod/blob/5c131cbf9956740ca35ad2aeccca68ea62d0e289/components/ws-daemon/pkg/cgroup/plugin_process_priority_v2.go#L130-L161)

Reply:
sounds good